contrib/openssl/async_engine.c

```diff
@@ -224,21 +241,24 @@ spawn_delay_thread()
 {
   pthread_t thread_id;
   OSSL_ASYNC_FD signal_fd;
-  OSSL_ASYNC_FD pipefds[2] = {0, 0};
   ASYNC_JOB *job;
-  if ((job = ASYNC_get_current_job()) == NULL)
+  if ((job = ASYNC_get_current_job()) == NULL) {
+      printf("Spawn no job\n");
       return;
+  }
 
   ASYNC_WAIT_CTX *waitctx = ASYNC_get_wait_ctx(job);
 
   size_t numfds;
   if (ASYNC_WAIT_CTX_get_all_fds(waitctx, &signal_fd, &numfds) && numfds > 0) {
```

Comment:
It is better to call "ASYNC_WAIT_CTX_get_all_fds(waitctx, nullptr, &numfds)" first.

Reply:
If we got an existed fd from waitctx, it means there is an async job still in progress. It should return immediately.